### PR TITLE
updated to v0.2.0 of SPGBox

### PR DESCRIPTION
**Pull request by lmiq**
please update SPGBox, the interface is now simpler and similar to those of other optimization methods

### CODE REVIEW

**Comment by rht** [comment]
Thank you!

File: pharmcat_v2_g.jl [hunk 1]
```diff
@@ -94,8 +94,7 @@ function do_optimize(fn, xs0, DeltaT, g!)
     #result = Optim.optimize(fn, g!, lower, upper, xs0, Optim.Fminbox(Optim.ConjugateGradient(linesearch = linesearch)), options)
 
     # Using SPGBox
-    spg_g!(x, g) = g!(g, x)
-    result = SPGBox.spgbox!(xs0, fn, spg_g!, l = lower, u = upper)
+    result = SPGBox.spgbox!(fn, g!, xs0, lower = lower, upper = upper)
```
Comment (rht) [inline pharmcat_v2_g.jl:97]: Additionally, I noticed that lower, upper are matched as positional arguments in the Optim.jl version.

```
result = Optim.optimize(fn, g!, lower, upper, xs0, Optim.Fminbox(Optim.ConjugateGradient(linesearch = linesearch)), options)
```

Comment (lmiq) [reply pharmcat_v2_g.jl:97]: Uhm.. I thought it was `(fn, g!, x0, lower, upper)` and implemented that. I will change that (and release a breaking 0.3.0 version later today).

Comment (lmiq) [reply pharmcat_v2_g.jl:97]: fixed in 0.3.0 which is released now

**Pull request merged**